File: CS/DevExpress.Calendar/PageLoadEventArgs.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraBars;

namespace DevExpress.Calendar {
    public class PageLoadEventArgs {
        public PageLoadEventArgs(Type typeInfo,
                    bool shouldLoadData, object state) {
            this._typeInfo = typeInfo;
            this._shouldLoadData = shouldLoadData;
            this._state = state;
        }

        public PageLoadEventArgs(Type typeInfo, object state) {
            this._typeInfo = typeInfo;
            this._shouldLoadData = true;
            this._state = state;
        }

        Type _typeInfo;
        public Type TypeInfo {
            get {
                return this._typeInfo;
            }
        }

        object _state;
        public object State {
            get {
                return this._state;
            }
        }

        bool _shouldLoadData = true;
        public bool ShouldLoadData {
            get {
                return _shouldLoadData;
            }
            set {
                _shouldLoadData = false;
            }
        }
    }
}

File: VB/DevExpress.Calendar/PageLoadEventArgs.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraBars

Namespace DevExpress.Calendar
	Public Class PageLoadEventArgs
		Public Sub New(ByVal typeInfo As Type, ByVal shouldLoadData As Boolean, ByVal state As Object)
			Me._typeInfo = typeInfo
			Me._shouldLoadData = shouldLoadData
			Me._state = state
		End Sub

		Public Sub New(ByVal typeInfo As Type, ByVal state As Object)
			Me._typeInfo = typeInfo
			Me._shouldLoadData = True
			Me._state = state
		End Sub

		Private _typeInfo As Type
		Public ReadOnly Property TypeInfo() As Type
			Get
				Return Me._typeInfo
			End Get
		End Property

		Private _state As Object
		Public ReadOnly Property State() As Object
			Get
				Return Me._state
			End Get
		End Property

		Private _shouldLoadData As Boolean = True
		Public Property ShouldLoadData() As Boolean
			Get
				Return _shouldLoadData
			End Get
			Set(ByVal value As Boolean)
				_shouldLoadData = False
			End Set
		End Property
	End Class
End Namespace
